### PR TITLE
Database encryption supporded for sqlite plugin

### PR DESCRIPTION
**Pull request by dryganets**

### CODE REVIEW

File: src/CordovaNativeSqliteProvider.ts [hunk 1]
```diff
@@ -18,6 +18,8 @@ export interface SqlitePluginDbOptionalParams {
     createFromLocation?: number;
     androidDatabaseImplementation?: number;
     androidLockWorkaround?: number;
+    // Database encryption pass phrase
+    key?: string;
```
Comment (deregtd) [inline src/CordovaNativeSqliteProvider.ts]: Call it encryptionKey instead?

File: src/CordovaNativeSqliteProvider.ts [hunk 2]
```diff
@@ -62,16 +64,27 @@ export class CordovaNativeSqliteProvider extends SqlProviderBase.SqlProviderBase
...
+            console.log('database ', dbName, ' opened successfuly');
+            task.resolve();
+        }, () => {
+            console.log('database ', dbName, ' open failed');
+            task.reject();
+        });
 
-        return this._ourVersionChecker(wipeIfExists);
+        return task.promise().then(() => {
+            console.log('promise then');
```
Comment (deregtd) [inline src/CordovaNativeSqliteProvider.ts]: remove logging

File: src/CordovaNativeSqliteProvider.ts [hunk 3]
```diff
@@ -62,16 +64,27 @@ export class CordovaNativeSqliteProvider extends SqlProviderBase.SqlProviderBase
...
             location: 2
-        }, this._openOptions));
+        }, this._openOptions);
 
-        if (!this._db) {
-            return SyncTasks.Rejected<void>('Couldn\'t open database: ' + dbName);
-        }
+        const task = SyncTasks.Defer<void>();
+        this._db = this._plugin.openDatabase(dbParams, () => {
+            console.log('database ', dbName, ' opened successfuly');
```
Comment (deregtd) [inline src/CordovaNativeSqliteProvider.ts]: remove logging

Comment (dryganets) [inline src/CordovaNativeSqliteProvider.ts]: ok

File: src/CordovaNativeSqliteProvider.ts (hunk 1 above)
Comment (dryganets) [inline src/CordovaNativeSqliteProvider.ts]: it's called key in sqlite-storage (at least it was so in commneted code)
would prefer to keep the original naming to simplify merge of sqlite-storage encription part.

**Pull request merged**